### DISCUSSION TITLE
services.AddFusionCache().WithOptions(fussionCacheOption) is not working

### DISCUSSION THREAD

**Discussion by pascalmartin** [ZiggyCreatures/FusionCache#327] [Q&A] [answered]
Helo 

Because the UseRegisteredOptions = true by default the builder will never user the Options set with the WithOptions method caussing a lot of investigation. Should the method WithOptions  automatiquely set the UseRegisteredOptions  = false;

Current workaroud is to force UseRegisteredOptions  = false;

```
var fusionCacheBuilder = services.AddFusionCache()
    .WithOptions(fusionCacheOptions)
    .WithNewtonsoftJsonSerializer();
fusionCacheBuilder.UseRegisteredOptions = false;
```

**Comment by jodydonetti** [comment, marked as answer]
Hi @pascalmartin , damn I think you're right!

I think it was never noticed because most people use the `options => { }` overload.

I'll create an issue and will fix it, thanks!

**Reply by pascalmartin** [reply]
Great, thank you very much

**Reply by jodydonetti** [reply]
Hi @pascalmartin  , [v2.0.0-preview-1](https://github.com/ZiggyCreatures/FusionCache/releases/tag/v2.0.0-preview-1) is out 🥳
This includes the fix to the bug you discovered, on top of the 2 mega fetures: Tagging and Clear() support!

🙏 Please, if you can try it out and let me know what you think, how it feels to use it or anything else really: any contribution is essential, thanks!

**Reply by pascalmartin** [reply]
Super thank i will try the tests soon

**Reply by jodydonetti** [reply]
I just released [preview-2](https://github.com/ZiggyCreatures/FusionCache/releases/tag/v2.0.0-preview-2), just in case you were doing some tests and wanted to try the very latest.

Thanks!

**Reply by pascalmartin** [reply]
I confirm the bug is fixe with the new version v2.0.0-preview-2
Upgrade to v2 without any breaking change and everything seems to be working fine
with
ZiggyCreatures.FusionCache 2.0.0-preview-2
ZiggyCreatures.FusionCache.Backplane.StackExchangeRedis 2.0.0-preview-1 (No 2.0.0-preview-2 available)
ZiggyCreatures.FusionCache.Serialization.NewtonsoftJson 2.0.0-preview-1  (No 2.0.0-preview-2 available)

Thanks a lot

**Reply by jodydonetti** [reply]
> I confirm the bug is fixe with the new version v2.0.0-preview-2
> Upgrade to v2 without any breaking change and everything seems to be working fine

Awesome 🥳

> with ZiggyCreatures.FusionCache 2.0.0-preview-2

Are you ok in using v2 previews until that will be GA or do you need an official v1.X ?

> ZiggyCreatures.FusionCache.Backplane.StackExchangeRedis 2.0.0-preview-1 (No 2.0.0-preview-2 available)
ZiggyCreatures.FusionCache.Serialization.NewtonsoftJson 2.0.0-preview-1 (No 2.0.0-preview-2 available)

Yep, I made a preview-2 only for the main lib, since nothing changed in the other satellite libs and they are compatible.

> Thanks a lot

Thanks to you!

**Reply by pascalmartin** [reply]
We are ok with the current 1.4.0 version with the small fix we do. When version 2 become GA we just find the TODO comment and remove the fix after upgrading to v2.

**Reply by jodydonetti** [reply]
Hi @pascalmartin , FYI [preview-3](https://github.com/ZiggyCreatures/FusionCache/releases/tag/v2.0.0-preview-3) just landed.

**Reply by jodydonetti** [reply]
Hi @pascalmartin , I forgot to ping you here: [v2.0.0](https://github.com/ZiggyCreatures/FusionCache/releases/tag/v2.0.0) is finally out which contains the fix to this 🎉
